### ISSUE TITLE
Building iOS from source fails with "-lReact-Codegen not found" error

### ISSUE THREAD

**Issue by gontovnik** [facebook/react-native#36016]
### Description

We have forked RN 0.70.6 to apply the fix discussed in https://github.com/facebook/react-native/issues/35936. When we try to run on iOS, we get the following error:
```
ld: library not found for -lReact-Codegen

clang: error: linked command failed with exit code 1 (use -v to see invocation)
```

### Version

0.70.6

### Output of `npx react-native info`

```
System:
    OS: macOS 12.6
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Memory: 40.81 MB / 64.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.11.0 - /usr/local/bin/node
    Yarn: 1.22.19 - ~/workspace/projects/apperry/anima-frontend/mobile/node_modules/.bin/yarn
    npm: 8.19.2 - /usr/local/bin/npm
    Watchman: 2022.10.31.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /Users/danilgontovnik/.rvm/gems/ruby-2.7.2/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    Android SDK:
      API Levels: 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 33
      Build Tools: 28.0.3, 29.0.2, 29.0.3, 30.0.2, 30.0.3, 31.0.0, 32.0.0, 33.0.0
      System Images: android-23 | Google APIs Intel x86 Atom, android-24 | Google Play Intel x86 Atom, android-26 | Google Play Intel x86 Atom, android-28 | Google Play Intel x86 Atom, android-29 | Google Play Intel x86 Atom, android-30 | Google APIs Intel x86 Atom, android-30 | Google Play Intel x86 Atom, android-33 | Google APIs ARM 64 v8a, android-33 | Google APIs Intel x86 Atom_64, android-33 | Google Play ARM 64 v8a, android-33 | Google Play Intel x86 Atom_64
      Android NDK: 22.1.7171670
  IDEs:
    Android Studio: 2021.2 AI-212.5712.43.2112.8512546
    Xcode: 14.2/14C18 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.16.1 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.1.0 => 18.1.0
    react-native: git+ssh://git@github.com:apperry/react-native-0.70.6-anrs.git#v0.70.6 => 0.70.6
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

### Steps to reproduce

1. Fork `react-native` from 0.70.6 tag
2. Add the forked version of RN as a dependency to your project
3. Run `yarn install && npx pod-install && yarn ios`

### Snack, code example, screenshot, or link to a repository

N/A

**Comment by gontovnik** [comment]
Update:

Looks like adding `rm -rf node_modules/react-native/packages/react-native-codegen/` to `postinstall` in `package.json` helps with the issue...

**Comment by cortinico** [comment]
> We have forked RN 0.70.6 to apply the fix discussed in #35936. When we try to run on iOS, we get the following error:

Closing as we're about to release 0.70.7 in the near future, we recommend you use that version instead.

**Comment by gontovnik** [comment]
@cortinico I can see the v0.70.7, and thanks a lot for that!

In regard to closing the issue – I think the issue of "building iOS from source" is not resolved and other folks who might want to fork iOS will face the same problems.
